src/main/java/Duke.java

```diff
@@ -1,10 +1,371 @@
+                + LINE_DIVIDER;
+        System.out.println(addedMessage);
+    }
+
+    public static void printAddedEvent(String eventDescription, String eventAt) {
+        String addedMessage = LINE_DIVIDER + System.lineSeparator()
+                + "Alright! I've successfully added this task:"  + System.lineSeparator()
+                + "[E]" + "[ " + "] " + eventDescription + "(at: " + eventAt + ")" + System.lineSeparator()
+                + "You now have " + tasks.size() + " tasks in the list!" + System.lineSeparator()
+                + LINE_DIVIDER;
+        System.out.println(addedMessage);
+    }
+
+    public static void printList() {
+        System.out.println(LINE_DIVIDER + System.lineSeparator()
+                + "As requested, here are the tasks in your list:");
+        if (tasks.size() == 0) {
+            System.out.println("There are no tasks recorded!");
+        } else {
+            for (Task element : tasks) {
+                System.out.println((tasks.indexOf(element) + ARRAYLIST_PRINT_OFFSET) + "." + element.toString());
+            }
+        }
+        System.out.println(LINE_DIVIDER);
+    }
+
+    public static int filterTaskNum(String doneTask) throws DukeMissingParamException, NumberFormatException {
+        String[] words = doneTask.split(" ");
+
+        if (words.length > 1) { // simple check to see if task number has not been input
+            return Integer.parseInt(words[1]);
+        }
+        else {
+            throw new DukeMissingParamException();
+        }
+    }
+
+    public static void markTaskDone(int numToMark) {
+        if ((numToMark - DONE_OFFSET >= 0) && (tasks.get(numToMark - DONE_OFFSET) != null)) {
+            tasks.get(numToMark - DONE_OFFSET).markAsDone();
+            System.out.println(LINE_DIVIDER + System.lineSeparator()
+                    + "Great work! I've marked this task as done:" + System.lineSeparator()
+                    + "[" + tasks.get(numToMark - DONE_OFFSET).getType() + "]" + "[" + tasks.get(numToMark - DONE_OFFSET).getStatusIcon() + "] "
+                    + tasks.get(numToMark - DONE_OFFSET).description + System.lineSeparator()
+                    + LINE_DIVIDER);
+        }
+        else {
+            System.out.println(DONE_INVALID_NUM);
+        }
+    }
+
+    public static void addTask(Task t) {
+        tasks.add(t);
+    }
+
+    public static void deleteTask(int numToRemove) throws DukeMissingParamException, NumberFormatException, IndexOutOfBoundsException {
+        if ((numToRemove - DELETE_OFFSET >= 0) && (tasks.get(numToRemove - DELETE_OFFSET) != null)) {
+            String byOrAt = "";
+            if (tasks.get(numToRemove - DELETE_OFFSET).getType().equals("D")) {
+                byOrAt = "by: ";
+            } else if (tasks.get(numToRemove - DELETE_OFFSET).getType().equals("E")) {
+                byOrAt = "at: ";
+            } else {
+                byOrAt = "";
+            }
```

Comment:
The use of a variable named "byOrAt" could be confusing. Perhaps you could refactor the code so that you can split into 2 variables "by" and "at" instead?